java/arcs/core/util/Json.kt

```diff
@@ -76,7 +76,33 @@ sealed class JsonValue<T>() {
 
     /** Used to represent a Json string */
     data class JsonString(override val value: String) : JsonValue<String>() {
-        override fun toString() = "\"${value.replace("\"", "\\\"")}\""
+        private val escapes = mapOf(
+            "\\" to "\\\\",
+            "\"" to "\\\"",
+            "\b" to "\\b",
+            12.toChar().toString() to "\\f",
+            "\n" to "\\n",
+            "\r" to "\\r",
+            "\t" to "\\t"
+        )
+
+        private fun escape(string: String): String {
+            var rtn = string
+            escapes.forEach { old, new ->
+                rtn = rtn.replace(old, new)
+            }
+            val regex = Regex("[^\u0000-\u007F]")
+            rtn = rtn.replace(regex) { match ->
+                var unicode = ""
+                match.value.toCharArray().forEach { char ->
+                    unicode = unicode + ("\\u${char.toInt()}")
```

Comment:
I think unicode escapes are hex, so shouldn't this been char.toInt().toString(16)?